### PR TITLE
isPrivacyBadgerEnabled made efficient #623

### PR DESCRIPTION
**Pull request by nextLane**
#623 
Converted disabledSites array to JavaScript object. 
Upgrade supported.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -210,12 +213,14 @@ var Utils = exports.Utils = {
    **/
   disablePrivacyBadgerForOrigin: function(origin){
     if(localStorage.disabledSites === undefined){
-      localStorage.disabledSites = JSON.stringify([origin]);
+      var obj = {};
+      obj[origin] = true;
+      localStorage.disabledSites = JSON.stringify(obj);
```
Comment (kaushik94) [inline src/utils.js]: You should never access `localStorage` naked. `setItem` and `getItem` are wrappers for this and they are the recommended way of doing. This is an excellent SO answer http://stackoverflow.com/questions/12632463/is-localstorage-getitemitem-better-than-localstorage-item-or-localstoragei/24904863#24904863.

File: src/utils.js [hunk 2]
```diff
@@ -36,6 +36,7 @@
 require.scopes["utils"] = (function() {
   
 var exports = {};
+var disabledSites = localStorage.disabledSites;
```
Comment (kaushik94) [inline src/utils.js]: `var disabledSites = localStorage.getItem("disabledSites");`

File: src/background.js [hunk 3]
```diff
@@ -112,10 +112,25 @@ function migrateVersion(prevVersion,currentVersion){
...
 
+
+/**
+ * migrates disabled sites from array data structure to object
+ */
+function migrateDisabledSites() {
+  
+   var obj = {};
+   var sites = JSON.parse(localStorage.disabledSites);
+  for(var i = 0; i < sites.length; i++){
```
Comment (kaushik94) [inline src/background.js]: Here are a few better ways.
http://stackoverflow.com/questions/26264956/convert-object-array-to-hash-map-indexed-by-an-attribute-value-of-the-object
http://stackoverflow.com/questions/28835262/convert-array-to-hashmap-key-value-pair-javascript

**Comment by nextLane** [comment]
Closed the PR for now, tests failed and the code needs some modifications. To address the issue of direct access to local storage as @kaushik94  rightly pointed out, opened the issue #685

**Pull request closed without merging**